src/state_machine.zig

```diff
@@ -339,20 +339,19 @@ pub const StateMachine = struct {
             const exists = insert.value_ptr.*;
             if (exists.debit_account_id != t.debit_account_id) {
                 return .exists_with_different_debit_account_id;
-            }
-            if (exists.credit_account_id != t.credit_account_id) {
+            } else if (exists.credit_account_id != t.credit_account_id) {
                 return .exists_with_different_credit_account_id;
-            }
-            if (exists.amount != t.amount) return .exists_with_different_amount;
-            if (@bitCast(u32, exists.flags) != @bitCast(u32, t.flags)) {
+            } else if (exists.amount != t.amount) {
+                return .exists_with_different_amount;
+            } else if (@bitCast(u32, exists.flags) != @bitCast(u32, t.flags)) {
                 return .exists_with_different_flags;
-            }
-            if (exists.user_data != t.user_data) return .exists_with_different_user_data;
-            if (!equal_32_bytes(exists.reserved, t.reserved)) {
+            } else if (exists.user_data != t.user_data) {
+                return .exists_with_different_user_data;
+            } else if (!equal_32_bytes(exists.reserved, t.reserved)) {
                 return .exists_with_different_reserved_field;
-            }
-            if (exists.timeout != t.timeout) return .exists_with_different_timeout;
-            return .exists;
+            } else if (exists.timeout != t.timeout) {
+                return .exists_with_different_timeout;
+            } else return .exists;
```

Comment:
Let's rather keep the `if` statements all separate as before because they're checking different fields.